cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewApplicationManager.java

```diff
@@ -70,7 +72,8 @@
                             UsageRegistry usageRegistry, ArtifactRepository artifactRepository,
                             AuthenticationContext authenticationContext, Impersonator impersonator,
                             AuthorizationEnforcer authorizationEnforcer, PluginFinder pluginFinder,
-                            CapabilityReader capabilityReader) {
+                            CapabilityReader capabilityReader,
+                            @Named("local") ConfiguratorFactory configuratorFactory) {
```

Comment:
No, the Configurators require a lot of parameters in their constructors. We dont have that info at this level to create a configurator object so we need to pass in the factory so that we can create the correct object when all the info is present.

The LocalArtifactLoaderStage is also used by the pipeline deployment where we need to have a factory to decide between remote and local implementations, so this class needs to use the same interface to keep things consistent